placeSpecificContent/hq/courtyard/controlPanel/controlPanel_server.js

```diff
@@ -0,0 +1,406 @@
+                            };
+                        }
+                    }
+                });
+            }
+            
+            if (Object.keys(BUTTON_TO_ENTITY_POSITIONING_MAP).length !== NUMBER_OF_BUTTONS) {
+                BUTTON_TO_ENTITY_POSITIONING_MAP = {};
+            }
+        },
+
+        /* PLAY A SOUND: Plays a sound at the specified position, volume, local mode, and playback 
+        mode requested. */
+        playSound: function(sound, volume, position, localOnly, loop) {
+            if (sound.downloaded) {
+                if (injector) {
+                    injector.stop();
+                    injector = null;
+                }
+                injector = Audio.playSound(sound, {
+                    position: position,
+                    volume: volume,
+                    localOnly: localOnly,
+                    loop: loop
+                });
+            }
+        },
+
+        /* Remotely called when the user presses a button. The id of the button is sent as a param and then used to 
+        match an index of the map to get target positions for each entity. Call another function to set up movement 
+        for each entity and then set the current button to transition color and all other buttons to inactive color.
+         Play the transition sound and edit the velocity of entities that will move.
+         */
+        pressButton: function(thisID, params) {
+            if (!ready) {
+                return;
+            }
+
+            courtyardInPlace = false;
+            roundtableInPlace = false;
+            courtyardVelocity = 0;
+            roundtableVelocity = 0;
+
+            var buttonID = params[0];
+            
+            var targetCourtyardPosition = BUTTON_TO_ENTITY_POSITIONING_MAP[buttonID].courtyardPosition;
+            var targetRoundtablePosition = BUTTON_TO_ENTITY_POSITIONING_MAP[buttonID].roundtablePosition;
+
+            _this.setUpMovement(roundtable, targetRoundtablePosition);
+            _this.setUpMovement(courtyard, targetCourtyardPosition);
+
+            Object.keys(BUTTON_TO_ENTITY_POSITIONING_MAP).forEach(function(button) {
+                if (button === buttonID) {
+                    Entities.editEntity(button, {
+                        backgroundColor: BUTTON_TRANSITION_COLOR,
+                        textColor: BUTTON_INACTIVE_COLOR
+                    });
+                } else {
+                    Entities.editEntity(button, {
+                        backgroundColor: BUTTON_INACTIVE_COLOR,
+                        textColor: BUTTON_TRANSITION_COLOR
+                    });
+                }
+            });
+
+            var soundPosition = Entities.getEntityProperties(courtyard, 'position').position;
+            _this.playSound(TRANSITION_SOUND, TRANSITION_VOLUME, soundPosition, false, true);
+            
+            if (!roundtableInPlace && roundtableVelocity) {
```

Comment:
when would `roundtableInPlace` ever be `true` here?
when would `roundtableVelocity` ever be falsey here?

Reply:
If the roundtable does not need to move, 'roundtableInPlace' gets set at line 334 or line 356 in the setUpMovement function. The roundtableVelocity is also set in this function.